src/collision.cpp

```diff
@@ -214,16 +214,50 @@ static bool add_area_node_boxes(const v3s16 min, const v3s16 max, IGameDef *game
 	thread_local std::vector<aabb3f> nodeboxes;
 	Map *map = &env->getMap();
 
+	const bool air_walkable = nodedef->get(CONTENT_AIR).walkable;
+
+	v3s16 last_bp(S16_MAX);
+	MapBlock *last_block = nullptr;
+
+	// Note: as the area used here is usually small, iterating entire blocks
+	// would actually be slower by factor of 10.
+
 	v3s16 p;
 	for (p.Z = min.Z; p.Z <= max.Z; p.Z++)
 	for (p.Y = min.Y; p.Y <= max.Y; p.Y++)
 	for (p.X = min.X; p.X <= max.X; p.X++) {
```

Comment:
General, somewhat unrelated question... Since we cache the y-block-location is MapSector, should the inner loop along nodes not always be Y?

Reply:
I thought so too but with the `ScopeProfiler` in place this actually regressed runtime heavily

Reply:
Interesting! And unexpected.